homeassistant/components/tado/climate.py

```diff
@@ -82,43 +53,88 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     for tado in api_list:
         for zone in tado.zones:
             if zone["type"] in [TYPE_HEATING, TYPE_AIR_CONDITIONING]:
-                entity = create_climate_entity(tado, zone["name"], zone["id"])
+                entity = create_climate_entity(hass, tado, zone["name"], zone["id"])
                 if entity:
                     entities.append(entity)
 
     if entities:
         add_entities(entities, True)
 
 
-def create_climate_entity(tado, name: str, zone_id: int):
+def create_climate_entity(hass, tado, name: str, zone_id: int):
     """Create a Tado climate entity."""
     capabilities = tado.get_capabilities(zone_id)
     _LOGGER.debug("Capabilities for zone %s: %s", zone_id, capabilities)
 
     zone_type = capabilities["type"]
+    support_flags = SUPPORT_PRESET_MODE | SUPPORT_TARGET_TEMPERATURE
+    supported_hvac_modes = [
+        TADO_TO_HA_HVAC_MODE_MAP[CONST_MODE_OFF],
+        TADO_TO_HA_HVAC_MODE_MAP[CONST_MODE_SMART_SCHEDULE],
+    ]
+    supported_fan_modes = None
+    heat_temperatures = None
+    cool_temperatures = None
 
-    ac_support_heat = False
     if zone_type == TYPE_AIR_CONDITIONING:
-        # Only use heat if available
-        # (you don't have to setup a heat mode, but cool is required)
         # Heat is preferred as it generally has a lower minimum temperature
-        if "HEAT" in capabilities:
-            temperatures = capabilities["HEAT"]["temperatures"]
-            ac_support_heat = True
-        else:
-            temperatures = capabilities["COOL"]["temperatures"]
-    elif "temperatures" in capabilities:
-        temperatures = capabilities["temperatures"]
+        for mode in ORDERED_KNOWN_TADO_MODES:
+            if mode in capabilities:
```

Comment:
Please use guard clauses to keep the code clean/reduce indentation.

```python
if mode not in capabilities:
    continue

supported_vac_modes.append(…)
```

Reply:
Fixing.  So much cleaner.  It's a pattern I need to get in habit of.